### DISCUSSION TITLE
Beta 1.27.10: Stops working after time/sleep with Monterey

### DISCUSSION THREAD

**Discussion by pictosun** [rxhanson/Multitouch-Community#224] [General]
Hi,

just have to give a short feedback concerning the beta I got.

Multitouch gestures will stop working after amount of time (maybe because of system sleep/close lid). 

If I open Multitouch settings it is working without any issue. But I have to open the settings.

**Comment by rxhanson** [comment]
Thanks for the report. This shouldn't be directly related to only the beta, but there is a bug with macOS that is related to timing with registering to listen for touch data upon wake, especially with deep sleep. It's likely that the gesture listening gets put in place once the app becomes frontmost. There are parameters for adjusting how long Multitouch will wait before restarting itself upon wake, but I'll make a follow up build that tries out something with toggling Multitouch into and out of frontmost upon wake, if you're up for trying that out. I can't reproduce the issue myself.

**Comment by pictosun** [comment]
Thx for the feedback. I will test it. Sure.

**Comment by pictosun** [comment]
Hi. Got an update to .11

But the issue is still there.

**Comment by rxhanson** [comment]
Thanks for trying it out and letting me know. This does come down to a timing issue, and I'll give it another go in the next release. Until then, if you want you can adjust the delay for starting the app with this terminal command:

`defaults write com.brassmonkery.Multitouch restartOnWakeDelay -int 5`

Let me know if you try it out and if that fixes it for you.

**Comment by pictosun** [comment]
Hi. 

I'm using v 1.27.11 (166) with newest Ventura and it does happen that sometimes after reboot Multitouch (even if start at boot is enabled) does not work. Have to disable/enable or restart it to get it working.

Another issue is, that it stops working (when switching networks). Have to go into app menu to get it working again.

Do you know something about those issues?

**Comment by rxhanson** [comment]
Hmm. This is the first I've heard of these issues. At a start, can you send me your config? You can find it at: ~/Library/Preferences/com.brassmonkery.Multitouch.plist

There's nothing in the app that is tied network activity (the only requests the app can make are for checking a license or checking for updates - neither of which have any relation to the rest of the functionality of the app), so I suspect that you have something else going on with your Mac that would be at the root of this.

**Comment by pictosun** [comment]
Just happened again. I do not have any clue why and how often it does happen. But now it just happened after wake up from sleep again. Had to open the app again.

**Comment by rxhanson** [comment]
Thanks for letting me know. 

Please try this terminal command if you haven't already, and let me know if you have the issue again.

`defaults write com.brassmonkery.Multitouch restartOnWakeDelay -int 5`

Then restart the app.

Since I can't reproduce this personally, the best place to start is if I know what you have configured, so that I know what isn't working properly.

These types of issues can be caused by other factors external to Multitouch, so it can be difficult to narrow things down, but I'll see what I can do. Are you still on Monterey?

**Comment by pictosun** [comment]
Hi. On actual Sonoma I do still have the issues, that Multitouch stops working at some time. I have to quit the app and restart it, to work again.

Are you working on those issues?

And in general, when can we expect a new release of Multitouch? You said long time ago, that you're redoing the GUI and other stuff... As it's a paid app it would be nice to get some updates :-)

**Comment by rxhanson** [comment]
Thanks for following up. I haven't been actively trying to figure this out, but don't mind looking into it again. Can you provide your configuration? You can find it at Help > Debug > Copy Debug Info to clipboard.

**Comment by pictosun** [comment]
Hi,
any update here? Still having issues within Sonoma.

Multitouch: 1.27.32 | 187
macOS: Version 14.6.1 (Build 23G93)
Model: Mac15,10

**Comment by rxhanson** [comment]
Sorry to hear you're still having this issue. Since I can't reproduce it with my usage of Multitouch, at the very least I would need to load in your configuration so I can see if I can reproduce it with what you have configured. 

You can copy it from the menu bar menu: Help > Debug > Copy Debug Info to clipboard.

**Comment by byalextran** [comment]
i'm also running into this issue with both sonoma and sequoia. i just did a clean install of sequoia this past weekend so have very limited apps running. the primary use-case i have for multitouch is to use it in conjunction with keyboard maestro. 

i used that terminal command you suggested and will report back. 

here's my debug log:

```
Multitouch: 1.27.41 (196)
macOS: Version 15.0.1 (Build 24A348)
Model: Mac14,2
Devices: ["unknown:288230376464310562:312598818","internalTrackpad2:504403158265495719:167"]
```

**Reply by rxhanson** [reply]
Thanks for providing details up front. Seems like you're just using Trackpad gestures? Let me know if that terminal command with the wake delay doesn't work. If it works sometimes but not all the time, then you might have to increase the delay.

**Reply by byalextran** [reply]
oh, that's leftover from when i was using a trackpad. but now i'm using a mouse so those gestures are no longer needed. do they impact anything? should i delete them? so far things are good, but still early as i only remember running into this issue maybe once a week (that i can remember).

**Reply by byalextran** [reply]
not sure if it's related, but i had it happen twice today. i'm using multitouch keyboard shortcuts for window management and those stopped working randomly until i re-launched multitouch. the weird thing is that the hyperkey (configured through multitouch) was still working, just not the keyboard shortcuts for window management.

**Reply by rxhanson** [reply]
Thanks for the update. When this happens, can you pull up the logging and we can hopefully see why the keyboard shortcuts aren't working? In the Multitouch menu, go to `Help > Debug > View Logging`, then try to execute a keyboard shortcut.

Also, perhaps I can figure it out faster if you upload the defaults plist that contains all of your Multitouch settings. You can find it at

```
~/Library/Preferences/com.brassmonkery.Multitouch.plist
```

That way I can load everything in and see if I can reproduce it or find any loose ends.
